### PR TITLE
Fixed analytics + added monitoring & analytics state

### PR DESCRIPTION
**Pull request by dlymonk**
## Overview

- Added analytics `setUserId` when inspectionId is passed as URL params.
- Added analytics and monitoring app state in common-ui-web component: `selectVehicleType`, `login`, `createInspection`.

## Checklist before requesting a review

- [x] I have updated the unit tests based on the changes I made
- [x] I have updated the docs (TSDoc / README / global doc) to reflect my changes
- [x] I have performed self-QA of my feature by testing the apps and packages and made sure that :
  - No regression or new bug has occurred
  - The acceptance criteria listed in the ticket are met
  - **Self-QA was made on both desktop and mobile**

### CODE REVIEW

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx [hunk 1]
```diff
@@ -123,6 +126,7 @@ export function PhotoCapture({
   ...initialCameraConfig
 }: PhotoCaptureProps) {
   useI18nSync(lang);
+  const { authToken, vehicleType, steeringWheel } = useMonkAppState();
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx:129]: I don't think we should use `useMonkAppState` in the `PhotoCapture` component :
- If the `vehicleType`, `steeringWheel` etc are defined in the `useMonkAppState`, it means that they already are set in the monitoring and analytics (thanks to `useAppStateMonitoring` and `useAppStateAnalytics`).
- I asked you to also set these value in the PhotoCapture component in the case that the user is specifically NOT using the `useMonkAppState` hook, but still using the `PhotoCapture` component.
- Of course the `PhotoCapture` component does not know the vehicle type and steering wheel position and it's fine, but at least it knows the inspection ID and the auth token.

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx [hunk 2]
```diff
@@ -140,6 +144,8 @@ export function PhotoCapture({
   const loading = useLoadingState();
   const addDamageHandle = useAddDamageMode();
   useComplianceAnalytics({ inspectionId, sights });
+  useAppStateMonitoring({ authToken, inspectionId, vehicleType, steeringWheel });
+  useAppStateAnalytics({ inspectionId });
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx:148]: And I think these two hooks should not be exported and only be used inside the `common` package

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts [hunk 3]
```diff
@@ -191,10 +191,11 @@ export function usePhotoCaptureSightState({
         alreadyTakenSights: alreadyTakenSights.length,
         totalSights: captureSights.length,
         sightSelected: notCapturedSights.length > 0 ? notCapturedSights[0].label : null,
+        captureCompleted: notCapturedSights.length === 0,
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:194]: Shouldn't we also update the `captureCompleted` property after each sight captured and not only when fetching the inspection?

Comment (dlymonkai) [reply packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:194]: We handle this part in this callback: 
``` js
const handleGalleryValidate = () => {
    startTasks()
      .then(() => {
        analytics.trackEvent('Capture Completed');
        analytics.setUserProperties({ captureCompleted: true });
        onComplete?.();
      })
      .catch((err) => {
        loading.onError(err);
        handleError(err);
      });
  };
  ```

**Pull request merged**